hive-metastore/src/main/java/org/apache/iceberg/hive/HMSParametersSetter.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.hive;
+
+import java.util.Set;
+import org.apache.hadoop.hive.metastore.api.Table;
+
+/** An interface to set the required HMS parameters to the table. */
+interface HMSParametersSetter {
```

Comment:
Iceberg table on Hive and Iceberg-view on Hive needs to set few parameters like metadatalocation, schema, etc. Few of the metrics needs to be fetched from TableMetadata/ViewMetadata. Since we don't have a common abstract class for both. This consumer can help to set the respective parameters with their own implementation .